### PR TITLE
#5920 feat: adds Grid storybook component

### PR DESCRIPTION
**Pull request by chriddell**
Implements https://github.com/wellcometrust/corporate/issues/5920

### CODE REVIEW

File: config/storybook/styles/_storybook-global.scss [hunk 1]
```diff
@@ -56,3 +56,13 @@ h6 {
     transform: translate3d(0, 64px, 0);
   }
 }
+
+// Grid
+.st-grid-cell-visualiser {
```
Comment (creido-welly) [inline config/storybook/styles/_storybook-global.scss:61]: 👍

Comment (chriddell) [reply config/storybook/styles/_storybook-global.scss:61]: Likely we'll have to split this file in to more partials as Storybook grows, `st-` seemed like a good starting point for namespacing 🤘🏻

File: src/components/Grid/Grid.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  const columnCount: 1 | 2 | 3 = select('column count', [1, 2, 3], 3);
+
+  /**
+   * Creates an array of GridCell components based
+   * on the given columnCount from the Storybook select knob.
+   *
+   * @return {array} - array of React Components
+   */
+  const createGridCells = () => {
+    const cells = [];
```
Comment (creido-welly) [inline src/components/Grid/Grid.stories.tsx:17]: Juicy titbit for yer, if you wanna "func programming" it up you could also do

```
const cells = Array.from(Array(columnCount), (x, index) => {
      const i = index + 1;

      return (
        <GridCell column={i} columnCount={columnCount}>
          <div className="st-grid-cell-visualiser">
            {`Grid Cell ${i} of ${columnCount}`}
          </div>
        </GridCell>
      );
    });
```

😬
drop 🎤

**Review by creido-welly — APPROVED** [review]

File: config/storybook/styles/_storybook-global.scss (hunk 1 above)
Comment (creido-welly) [reply config/storybook/styles/_storybook-global.scss:61]: perhaps, all good for now tho

File: src/components/Grid/Grid.stories.tsx (hunk 2 above)
Comment (chriddell) [reply src/components/Grid/Grid.stories.tsx:17]: What do you think about `const i` not being type-checked? It throws this error:

```bash
Type 'number' is not assignable to type '1 | 2 | 3'
```

And trying to set explicit type on `i` (of `1 | 2 | 3`), e.g:

```javascript
const i: 1 | 2 | 3 = index + 1;
```

throws:

```bash
Type 'number' is not assignable to type '1 | 2 | 3'
```

Can't for the life of me figure out how to type-check it properly 😶

Comment (creido-welly) [reply src/components/Grid/Grid.stories.tsx:17]: have you tried `const i: number = index + 1;` ?

Comment (chriddell) [reply src/components/Grid/Grid.stories.tsx:17]: (realise we spoke about this IRL but just for others' clarity...) 

This doesn't work as the `column` property on `GridCell` is type-checked as `1 | 2 | 3`, so `i` must explicitly match that. 

I'll work on it a little more later.

**Pull request closed without merging**